### PR TITLE
Feature 2 : Delete ToDo functionality implemented.

### PR DESCRIPTION
**Pull request by saa-wtag**
[Task-2: Delete Todo](https://trello.com/c/xgWFRc6K)
This PR introduces the functionality of delete a todo by clicking the delete button beside input box.

**Functionalities:**
 Todo can be deleted
 A list is populated with the updated Todo

Please have a close look on this PR and kindly put your valuable comments.

https://github.com/sad-wtag/ToDoApp_VanillaJS/assets/164306406/d68f2713-9c73-4117-b0d3-92719aac97c5

### CODE REVIEW

**Review by aka-wtag — APPROVED** [review]
The formatting is not consistent. You can use the prettier extension. Thank you.

File: js/script.js [hunk 1]
```diff
@@ -6,16 +6,44 @@ let tasks = [];
...
-    $tasksList.textContent = taskTitle;
-    $taskList.prepend($tasksList);
-    $taskInput.value = "";
+    createTask(taskTitle);
     showToastMessage("Task added successfully!");
     $addButton.disabled = true;
   }
 };
 
+const deleteHandler = (taskId) => {
```
Comment (ray-wtag) [inline js/script.js]:
```suggestion
const deleteTask = (taskId) => {
```

File: js/script.js [hunk 2]
```diff
@@ -6,16 +6,44 @@ let tasks = [];
...
+};
+
+const renderTasks = () => {
+  $taskList.innerHTML = "";
+
+  tasks.forEach((task) => {
+    const tasksList = document.createElement("li");
+    tasksList.textContent = task.title;
+
+    const deleteButton = document.createElement("button");
```
Comment (ray-wtag) [inline js/script.js]: We can modularize these elements which would be created.

File: js/script.js [hunk 3]
```diff
@@ -6,16 +6,44 @@ let tasks = [];
...
+  renderTasks();
+
+  $taskInput.value = "";
+};
+
+const renderTasks = () => {
+  $taskList.innerHTML = "";
+
+  tasks.forEach((task) => {
+    const tasksList = document.createElement("li");
```
Comment (ray-wtag) [inline js/script.js]: ditto

**Review by ray-wtag — CHANGES_REQUESTED** [review]

File: js/script.js (hunk 2 above)
Comment (saa-wtag) [reply js/script.js]: Resolved.

File: js/script.js (hunk 3 above)
Comment (saa-wtag) [reply js/script.js]: Resolved.

File: js/script.js (hunk 1 above)
Comment (saa-wtag) [reply js/script.js]: Renamed.

File: js/script.js [hunk 4]
```diff
@@ -17,6 +17,11 @@ const addButtonHandler = () => {
   } else showToastMessage("Please provide a valid title!");
 };
 
+const deleteTask = (taskId) => {
+  tasks = tasks.filter((task) => task.id !== taskId);
```
Comment (maa-wtag) [inline js/script.js:21]:
```suggestion
  updatedTasks = tasks.filter((task) => task.id !== taskId);
```

**Review by maa-wtag — APPROVED** [review]

**Review by ray-wtag — APPROVED** [review]

File: js/script.js [hunk 5]
```diff
@@ -35,6 +40,10 @@ const renderTasks = () => {
   tasks.forEach((task) => {
     const texts = [task.title, " Created at: " + task.createdAt];
     const $tasksList = createElement(texts, "li");
+    const $deleteButton = createElement("Delete", "button", () =>
```
Comment (hhu-wtag) [inline js/script.js]: isn't every delete button supposed to be a part of the task itself? Create the button inside each task and then simply re-render the list.

**Review by hhu-wtag — CHANGES_REQUESTED** [review]

File: js/script.js (hunk 5 above)
Comment (saa-wtag) [reply js/script.js]: Done, bhai.

File: js/script.js [hunk 6]
```diff
@@ -17,13 +17,20 @@ const addButtonHandler = () => {
...
+  tasks = tasks.filter((task) => task.id !== taskId);
+  renderTasks();
+};
+
 const createTask = (taskTitle) => {
   const task = {
     id: new Date().getTime(),
     title: taskTitle,
     createdAt: formatDate(new Date()),
+    deleteButton: createElement("Delete", "button", () => deleteTask(task.id)),
```
Comment (hhu-wtag) [inline js/script.js]: deleteButton is not part of the `task` data structure. It's a part of the `task` component.

The idea is:
```js
function createTaskElement(task){
  const $task = createElement('div')
  const $text = createElement('p')
  const $deleteButton = createElement('button')
  
  $text.textContent = task.text
  $deleteButton.addEventListener(someEventListener)
  
  $task.appendChild($text)
  $task.appendChild($deleteButton)

  return $task
}

// script.js
$taskList.innerHTML = ""

tasks.forEach(task => {
  const $task = createTaskElement(task)
  
  $taskList.appendChild($task)
})
```

**Review by hhu-wtag — CHANGES_REQUESTED** [review]

File: js/script.js (hunk 6 above)
Comment (saa-wtag) [reply js/script.js]: resolved, bhai.

**Review by hhu-wtag — APPROVED** [review]

**Pull request merged**